### PR TITLE
build2slack: script to send build results to slack

### PR DESCRIPTION
**Pull request by 7AC**
Invoke from a post-build script with:

```
usage: build2slack.py [-h] --result {SUCCESS,FAILURE}
```

### CODE REVIEW

File: build2slack.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import yaml
+
+
+GERRIT_CHANGE_COMMIT_MESSAGE_ENV = "GERRIT_CHANGE_COMMIT_MESSAGE"
+GERRIT_CHANGE_NUMBER_ENV = "GERRIT_CHANGE_NUMBER"
+GERRIT_CHANGE_URL_ENV = "GERRIT_CHANGE_URL"
+GERRIT_PATCHSET_NUMBER_ENV = "GERRIT_PATCHSET_NUMBER"
+GERRIT_REVISION_ENV = "GERRIT_PATCHSET_REVISION"
+PACKAGE_SETTINGS_FILE = 'package.yml'
+SLACK_WEBHOOK_ENV = "SLACK_WEBHOOK"
```
Comment (tsuna) [inline build2slack.py]: Honestly I see zero benefit to having a named constant that is named with the value of the constant itself.

File: build2slack.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+def gerritRevision():
+  rev = getEnv(GERRIT_REVISION_ENV)
+  diff = subprocess.check_output(["git", "diff-tree", "--no-commit-id", "--name-only", "-r", rev])
+  return diff.split("\n")
+
+
+def dirs(files):
+  return set([os.path.dirname(f) for f in files])
```
Comment (tsuna) [inline build2slack.py]: Inner `[` and `]` are not necessary and should be removed.

File: build2slack.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+SLACK_WEBHOOK_ENV = "SLACK_WEBHOOK"
+
+
+def getEnv(name):
+  value = os.environ.get(name)
+  assert value, name + " is not set"
+  return value
+
+
+def gerritRevision():
```
Comment (tsuna) [inline build2slack.py]: Kinda weird for a function called `gerritRevision()` to return the list of affected files.

File: build2slack.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  payload = {"username": "jenkins",
+             "icon_url": "https://a.slack-edge.com/205a/img/services/jenkins-ci_72.png"}
+  payload["attachments"] = [{"color": "danger" if failed else "good",
+                             "title": title,
+                             "title_link": titleLink,
+                             "text": message,
+                             "mrkdwn_in": ["text"]}]
+  files = gerritRevision()
+  webHook = os.environ.get(SLACK_WEBHOOK_ENV)
+  assert webHook, SLACK_WEBHOOK_ENV + " is not set"
```
Comment (tsuna) [inline build2slack.py]: use `getEnv()` here?

**Review by tsuna — COMMENTED** [review]
Thanks for doing this!

File: build2slack.py (hunk 1 above)
Comment (7AC) [reply build2slack.py]: I agree, I added it initially as a note to self to remember what they were all called

File: build2slack.py (hunk 3 above)
Comment (7AC) [reply build2slack.py]: I'll rename it

File: build2slack.py (hunk 2 above)
Comment (7AC) [reply build2slack.py]: 👍

File: build2slack.py (hunk 4 above)
Comment (7AC) [reply build2slack.py]: Oh missed that, thanks

**Review by jfrabaute — APPROVED** [review]

**Pull request merged**